### ISSUE TITLE
make: lwt error

### ISSUE THREAD

**Issue by mootjes** [tezos/tezos#150]
While building on the raspberry pi, I received an error on build:
-) Here is my history: https://www.hastebin.com/dixutefafi.sql
-) make build-deps looked OK, except for one warning: https://www.hastebin.com/ixayuqolis.sql

[WARNING] Rsync partially failed:
symlink has no referent:
"/home/mootjes/tezos/.github/ISSUE_TEMPLATE.md"
rsync error: some files/attrs were not transferred (see previous
errors) (code
23) at main.c(1178) [sender=3.1.2]

-) make results in an error ￼
https://www.hastebin.com/haziwinehi.coffeescript

Detail:
File "src/utils/lwt_dropbox.ml", line 54, characters 23-44:
Warning 3: deprecated: Lwt.waiter_of_wakener
 This function should be avoided, because it makes subtyping of resolvers
 unsound. See
  https://github.com/ocsigen/lwt/issues/458
File "src/utils/lwt_dropbox.ml", line 1:
Error: Some fatal warnings were triggered (1 occurrences)

Any thoughts?

**Comment by cleanunicorn** [comment]
My errors look a bit different 

```
    ocamlopt lib_error_monad/tezos_error_monad.{a,cmxa}
      ocamlc lib_base/tezos_base__TzPervasives.{cmi,cmti}
    ocamlopt lib_stdlib_lwt/tezos_stdlib_lwt__Lwt_exit.{cmx,o}
      ocamlc lib_micheline/tezos_micheline__Micheline.{cmi,cmti}
      ocamlc lib_node_p2p_base/tezos_node_p2p_base__P2p_types.{cmi,cmti}
      ocamlc lib_protocol_compiler/main_embedded_packer.{cmi,cmti}
    ocamlopt lib_stdlib_lwt/tezos_stdlib_lwt__Lwt_utils.{cmx,o} (exit 2)
(cd _build/default && /home/daniel/.opam/tezos/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -w -9-30 -open Tezos_stdlib -open Tezos_data_encoding -open Tezos_error_monad -safe-string -g -I /home/daniel/.opam/tezos/lib/bytes -I /home/daniel/.opam/tezos/lib/cstruct -I /home/daniel/.opam/tezos/lib/ipaddr -I /home/daniel/.opam/tezos/lib/ipaddr/unix -I /home/daniel/.opam/tezos/lib/lwt -I /home/daniel/.opam/tezos/lib/ocaml -I /home/daniel/.opam/tezos/lib/ocaml/threads -I /home/daniel/.opam/tezos/lib/ocplib-endian -I /home/daniel/.opam/tezos/lib/ocplib-json-typed -I /home/daniel/.opam/tezos/lib/re -I /home/daniel/.opam/tezos/lib/result -I /home/daniel/.opam/tezos/lib/sexplib -I /home/daniel/.opam/tezos/lib/sexplib/0 -I /home/daniel/.opam/tezos/lib/stringext -I /home/daniel/.opam/tezos/lib/uri -I lib_data_encoding -I lib_error_monad -I lib_stdlib -no-alias-deps -I lib_stdlib_lwt -open Tezos_stdlib_lwt -o lib_stdlib_lwt/tezos_stdlib_lwt__Lwt_utils.cmx -c -impl lib_stdlib_lwt/lwt_utils.ml)
File "lib_stdlib_lwt/lwt_utils.ml", line 90, characters 8-29:
Warning 3: deprecated: Lwt.waiter_of_wakener
 This function should be avoided, because it makes subtyping of resolvers
 unsound. See
  https://github.com/ocsigen/lwt/issues/458
File "lib_stdlib_lwt/lwt_utils.ml", line 119, characters 8-29:
Warning 3: deprecated: Lwt.waiter_of_wakener
 This function should be avoided, because it makes subtyping of resolvers
 unsound. See
  https://github.com/ocsigen/lwt/issues/458
File "lib_stdlib_lwt/lwt_utils.ml", line 1:
Error: Some fatal warnings were triggered (2 occurrences)
Makefile:5: recipe for target 'all' failed
make: *** [all] Error 1

```

**Comment by maxtez-raspbaker** [comment]
the error is the same, but I think you are compiling different branches of Tezos, mootjes is going for the alphanets. The file "lwt_utils.ml" is not quite the same, in both branches it includes the "deprecated: Lwt.waiter_of_wakener" but in different parts of the code. What puzzles me is that I did compile both branches without this problem popping up and "Lwt.waiter_of_wakener" is well included. I am wondering if somehow you are using OCaml 4.05.0 instead of the recommended 4.04.2 for Tezos.

**Comment by christian-lefty** [comment]
I can confirm that I have the same error on 4.04.2

**Comment by maxtez-raspbaker** [comment]
this has been solved (the "good" versions for tezos are lwt.3.0.0 or lwt.3.1.0). 
See more info here:
https://forums.tezos.community/t/step-by-step-setup-of-a-raspberry-pi-3-with-fedora-linux-os-and-tezos/913

**Comment by demodun** [comment]
The Depracated function that causes compilation errors exist also elsewhere in your code
```
pwd
/tezos/src
grep -rn waiter_of_wakener
node/shell/net_validator.ml:341:  Lwt.protected (Lwt.waiter_of_wakener bootstrapped_wakener)
node/shell/distributed_db_functors.ml:228:                wrap s k ?timeout (Lwt.waiter_of_wakener data.wakener)
node/shell/distributed_db_functors.ml:234:        wrap s k ?timeout (Lwt.waiter_of_wakener data.wakener)
utils/lwt_utils.ml:214:        Lwt.waiter_of_wakener u
utils/lwt_utils.ml:243:        Lwt.waiter_of_wakener u
utils/lwt_dropbox.ml:54:        Lwt.protected (Lwt.waiter_of_wakener w)
environment/v1/lwt.mli:292:(* val waiter_of_wakener : 'a u -> 'a t *)

**Comment by maxtez-raspbaker** [comment]
I am not a developer of the code, just a user trying to run it. From a user perspective the workaround is to use lwt.3.1.0 or lwt.3.0.0 until the development team updates the Tezos code.
my apologies for the previous misleading statement.

**Comment by MiloDavis** [comment]
This is fixed as of 58ad60f38a4c4f9c084f03d04e57eef497648abd.
